esp-link/main.c

```diff
@@ -79,7 +79,19 @@ static void ICACHE_FLASH_ATTR prHeapTimerCb(void *arg) {
 }
 #endif
 
+# define VERS_STR_STR(V) #V
+# define VERS_STR(V) VERS_STR_STR(V)
+char* esp_link_version = VERS_STR(VERSION);
+
 void user_rf_pre_init(void) {
+  system_set_os_print(DEBUG_SDK);
+#if defined(STA_SSID) && defined(STA_PASS)
+  struct station_config stconf;
+  os_strncpy((char*)stconf.ssid, VERS_STR(STA_SSID), 32);
+  os_strncpy((char*)stconf.password, VERS_STR(STA_PASS), 64);
+  stconf.bssid_set = 0;
+  wifi_station_set_config_current(&stconf);
+#endif
```

Comment:
Other option could be to get the value from the flash wifi config and if it's not the same as the values in STA_PASS or STA_SSID then write the new values to flash.  This would remove any ambiguity.